src/libraries/Common/src/Interop/Browser/Interop.Runtime.Bridge.cs

```diff
@@ -0,0 +1,381 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+using JSObject = Interop.JavaScript.JSObject;
+using JSException = Interop.JavaScript.JSException;
+
+internal static partial class Interop
+{
+    internal static partial class Runtime
+    {
+        private static Dictionary<int, JSObject?> _boundObjects = new Dictionary<int, JSObject?>();
+        private static Dictionary<object, JSObject?> _rawToJS = new Dictionary<object, JSObject?>();
+
+        internal static int BindJSObject(int jsId, Type mappedType)
+        {
+            if (!_boundObjects.TryGetValue(jsId, out JSObject? obj))
```

Comment:
In general in .NET, static methods need to be safe for concurrent access; these aren't, given that they're accessing mutable dictionaries without synchronization.

Is the idea that it's ok to do so here because these interop types are only being used in a single-threaded environment?  Is it possible that invariant might be relaxed in the future?  If these may be used concurrently, this code needs to be updated.

Reply:
Yes, you are right.  There is another PR that has not made it into the main mono repo yet.  I will make those changes here.

Reply:
Addressed in recent commits